### ISSUE TITLE
Z-index should be re-calculated for already opened modals

### ISSUE THREAD

**Issue by doubleOrt** [Dogfalo/materialize#4925]
Using _Materialize v0.99.0_

I suppose the problem is with how i am using materialize, but i am using modals like how activities are used in Android, so for example i have a modal that opens a user profile, on the user's profile i open another modal, say that user's posts, then i click on the user's name in that user posts modal, the user modal, which should now appear, does not, because the z-index is not recalculated. I could modify the z-index, but that would destroy the whole architecture for the modals.

I don't know what the disadvantages of re-calculating the z-index each time a modal is opened would be.

Here is how i am using modals (the whole screen is a single div with a modal class in each of these pictures): 

![localhost-ortify-logged_in html iphone 6](https://user-images.githubusercontent.com/18074909/27765529-4dc5db9e-5ebc-11e7-8a98-7be18ccc29bb.png)
![localhost-ortify-logged_in html iphone 6 1](https://user-images.githubusercontent.com/18074909/27765528-4dc57d70-5ebc-11e7-8204-b485f4bf555a.png)

**Comment by acburst** [comment]
Is there anyway you could recreate the problem in a simple codepen? I'm not sure exactly what the problem is. We support nesting modals

**Comment by doubleOrt** [comment]
@acburst I fixed my problem by overwriting all of Materialize's z-index calculations and modal openings. 
Steps to reproduce this issue: 
1- Create a modal with the id #modal1 
2- Inside of #modal1, create a modal-trigger for another modal, say #modal2.
3- After you go to #modal2 with #modal1's modal-trigger, #modal1 is still opened but its z-index is smaller than #modal2's so it is not visible. 
4- Now if you have a modal-trigger for #modal1 inside of #modal2, when you click it, nothing happens, because the z-index is not recalculated (since Materialize thinks #modal1 is already open so the methods just return or something). 

So basically, create a modal, go to another modal with a button from that initial modal, and then inside of that second modal try to go back to the first modal using another button, the initial modal, which logically should appear, does not.

**Comment by doubleOrt** [comment]
@acburst currently on a deadline so i can't spare any time to reproduce the issue with a Codepen, i created this issue out of desperation, but i fixed it myself after 6 hours of continuous work. What i did was i created  a stack for the modals and a custom z-index counter, so on a modal-trigger being clicked, i checked if it was already opened and if it was, i updated its z-index so it was on top, and i use that same stack on the modals being closed, so everything is working alright so far.

**Comment by doubleOrt** [comment]
@acburst  If this already has a solution, please inform me because i don't want to mingle with your defaults.

**Comment by doubleOrt** [comment]
@acburst  Okay, here is another issue i want to ask you about since you appear to be active in this library's issues section.
Is it mandatory for the select inputs to be recreated over and over again each time you re-initialize them, does this behavior persist in the newer versions of Materialize ? say i initialized a select input, so now everything is great, however i copy the HTML of the div containing it to another div, and now the select inputs, and their UL elements, generated by Materialize, will be copied as well (visible the source code), however those won't be actually shown anymore when i click their select input elements. I have to re-initialize them, and if i do that, then i end up with duplicate select inputs. 
So is there a solution for this or do i have to remove the already-generated UL elements manually before re-initializing. 

Sorry for the too-many questions, i feel like i am demanding too much of this library and i am using it for   something it was not meant to do (a mobile-app, virtually). And i did end up having to do the most things Materialize was supposed to do for me myself, like the tabs, even their animations didn't work as expected when they were added dynamically, that is the only issue with this library, it is not as dynamic as it should be. Perhaps  you guys could work on that.

**Comment by doubleOrt** [comment]
@acburst Concerning my last question (the one about select inputs), i just re-checked the documentation, you already have a function for destroying selects, thanks!!!! Will try that out and check if it is what i need.

**Comment by doubleOrt** [comment]
@acburst Code to reproduce the second bug (i am using the latest release, _Materialize v0.99.0_):  

HTML: 
`<div>

<select>
<option value="female">Female</option>
<option value="male">Male</option>
<option value="other">Other</option>
</select>

</div>`

Javascript:
 `$(document).ready(function(){
$("select").material_select();	
var old_html = $("div").html();
$("div").html("");
$("div").html(old_html);	// at this point, the old material_select has a buggy effect, where the select dropdown is rendered, but clicking it does not show the UL element.
$("select").material_select("destroy");	// Naturally, the solution that would cross one's mind at this point would be to destroy and re-initialize the select. However, "destroy" does not destroy the select element that we described above, material_select will work, and therefore we end up with duplicates.
$("select").material_select();		
});`

**Comment by acburst** [comment]
Copying generated html like this is not recommended. It will cause the references to be unloaded so the event handlers no longer work. Instead you should try destroying the select before you copy and overwrite the HTML or just remove the HTML and build the initial select from scratch and initialize that.

**Comment by tomscholz** [comment]
Closed due inactivity. Feel free to reopen it, if it's still necessary.
